boards/arm/nrf52840dongle_mdk/nrf52840dongle_mdk_defconfig

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_NRF52X=y
+CONFIG_SOC_NRF52840_QIAA=y
+CONFIG_BOARD_NRF52840DONGLE_MDK=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# Enable hardware stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# additional board options
+CONFIG_GPIO_AS_PINRESET=y
```

Comment:
The reset is needed to flash the board. The button has to be pressed to get in to the flash mode.

Reply:
Okay, so this option should stay, but then the definition of `button0` (and the `sw0` alias) is confusing. It suggests that there is a user button available to be used in applications; one could for example build the [`basic/button`](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/basic/button) sample for the board, but it won't work as expected, because the button will reset the board.

Reply:
You are wright. I will change the name & label to reset_button